From 5a6ff9ffba4f547dc48ba9bb80da409121325d03 Mon Sep 17 00:00:00 2001
From: Spyros Trigazis <strigazi@gmail.com>
Date: Fri, 12 Jan 2018 09:56:50 +0000
Subject: [PATCH] Use sort -V to compare golang versions

Change hack/lib/golang.sh to compare golang
version properly with "sort -V" and not as strings.
---
 hack/lib/golang.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hack/lib/golang.sh b/hack/lib/golang.sh
index 3e12b3170a..0e33119e0b 100755
--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -323,7 +323,7 @@ EOF
   go_version=($(go version))
   local minimum_go_version
   minimum_go_version=go1.9.1
-  if [[ "${go_version[2]}" < "${minimum_go_version}" && "${go_version[2]}" != "devel" ]]; then
+  if [[ "${minimum_go_version}" == $(echo -e "${minimum_go_version} ${go_version[2]}" | sort --version-sort | head -n1) && "${go_version[2]}" != "devel" ]]; then
     kube::log::usage_from_stdin <<EOF
 Detected go version: ${go_version[*]}.
 Kubernetes requires ${minimum_go_version} or greater.
-- 
2.15.1

